wandb/sdk/wandb_stream_table.py

```diff
@@ -0,0 +1,175 @@
+        self._entity_name = self._lite_run._entity_name
+        self._ensure_remote_initialized()
+        atexit.register(self._at_exit)
+
+    def _ensure_remote_initialized(self) -> None:
+        with self._lock:
+            self._lite_run.ensure_run()
+            print_url = False
+            self._artifact = self._stream_table_artifact()
+            if print_url:
+                base_url = wandb_public_api().settings["base_url"]
+                if base_url.endswith("api.wandb.ai"):
+                    base_url = base_url.replace("api", "weave.")
+                url = f"{base_url}/?exp=get%28%0A++++%22wandb-artifact%3A%2F%2F%2F{self._entity_name}%2F{self._project_name}%2F{self._table_name}%3Alatest%2Fobj%22%29%0A++.rows"
+                printer = get_printer(_get_python_type() != "python")
+                printer.display(
+                    f'{printer.emoji("star")} View data at {printer.link(url)}'
+                )
+
+    def _stream_table_artifact(self) -> "Artifact":
+        if self._artifact is None:
+            self._artifact = Artifact(
+                self._table_name,
+                type="stream_table",
+                metadata={
+                    "_weave_meta": {
+                        "is_panel": False,
+                        "is_weave_obj": True,
+                        "type_name": "stream_table",
+                    },
+                },
+            )
+            with self._artifact.new_file("obj.object.json") as f:
+                payload = {
+                    "_type": "stream_table",
+                    "table_name": self._table_name,
+                    "project_name": self._project_name,
+                    "entity_name": self._entity_name,
+                }
+                f.write(json.dumps(payload))
+            with self._artifact.new_file("obj.type.json") as f:
+                payload = {
+                    "type": "stream_table",
+                    "_base_type": {"type": "Object"},  # type: ignore[dict-item]
+                    "_is_object": True,  # type: ignore[dict-item]
+                    "table_name": "string",
+                    "project_name": "string",
+                    "entity_name": "string",
+                }
+                f.write(json.dumps(payload))
+            self._lite_run.log_artifact(self._artifact)
+        return self._artifact
+
+    def log(self, row_or_rows: ROW_TYPE) -> None:
+        if isinstance(row_or_rows, dict):
+            row_or_rows = [row_or_rows]
+
+        for row in row_or_rows:
+            self._log_row(row)
+
+    def rows(self) -> None:
+        raise errors.Error(
+            "reading stream tables is not supported in wandb, use weave.StreamTable"
+        )
+
+    def _log_row(self, row: dict) -> None:
+        row_copy = {**row}
+        if wandb.run is not None:
+            row_copy["_run"] = wandb.run.path
+        row_copy["_client_id"] = self._client_id
+        self._lite_run.log(row_copy)
```

Comment:
in my mind, this is where the serialization logic might go

Reply:
As mentioned above, I was imagining we add serialization logic to each `MediaType` via a `to_weave` or some duck typed conversion logic.